cupy/core/raw.pyx

```diff
@@ -122,10 +129,18 @@ cdef class RawKernel:
                 'translate_cucomplex': self.translate_cucomplex,
                 'file_path': self.file_path,
                 'name_expressions': self.name_expressions,
-                'enable_cooperative_groups': self.enable_cooperative_groups}
+                'enable_cooperative_groups': self.enable_cooperative_groups,
+                'raw_ver': self.raw_ver}
         return args
 
     def __setstate__(self, dict args):
+        raw_ver = args.get('raw_ver')
+        if raw_ver != 1:
+            raise pickle.UnpicklingError(
+                'The pickled RawKernel object is not supported by the current '
+                'CuPy version. It should not be used. Please recompile.')
+        self.raw_ver = raw_ver
```

Comment:
How about this implementation?
```suggestion
        if args.get('raw_ver') != self.raw_ver:
            raise pickle.UnpicklingError(
                'The pickled RawKernel object is not supported by the current '
                'CuPy version. It should not be used. Please recompile.')
```

Reply:
Done in 3f7eb27. Note that `raw_ver` has to be set early (in `__cinit__()`) in order to make this possible.

Reply:
I meant that I want to add `_` prefix to the variable name, but the implementation of 3f7eb27 is also OK!